diff --git arch/arm64/boot/dts/freescale/imx8mp-evk-usdhc1-m2.dts arch/arm64/boot/dts/freescale/imx8mp-evk-usdhc1-m2.dts
index 397a60436c2e..2add29f3ad19 100644
--- arch/arm64/boot/dts/freescale/imx8mp-evk-usdhc1-m2.dts
+++ arch/arm64/boot/dts/freescale/imx8mp-evk-usdhc1-m2.dts
@@ -17,11 +17,13 @@ reg_usdhc1_vmmc: regulator-usdhc1 {
 		enable-active-high;
 	};
 
-	usdhc1_pwrseq: usdhc1_pwrseq {
-		compatible = "mmc-pwrseq-simple";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_usdhc1_pwrseq>;
-		reset-gpios = <&gpio2 10 GPIO_ACTIVE_LOW>;
+	wlan_en: regulator-100 {
+		compatible = "regulator-fixed";
+		regulator-name = "wlan_en";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		enable-active-high;
+		gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
 	};
 };
 
@@ -29,6 +31,7 @@ &iomuxc {
 	pinctrl_reg_usdhc1_vmmc: regusdhc1vmmcgrp {
 		fsl,pins = <
 			MX8MP_IOMUXC_SD1_DATA4__GPIO2_IO06	0x140
+			MX8MP_IOMUXC_SD1_DATA7__GPIO2_IO09  0x000
 		>;
 	};
 
@@ -86,22 +89,30 @@ &pcie_phy {
 };
 
 &usdhc1 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1>;
-	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
+	bus-width = <4>;
 	keep-power-in-suspend;
+	cap-power-off-card;
 	non-removable;
 	wakeup-source;
-	mmc-pwrseq = <&usdhc1_pwrseq>;
-	vmmc-supply = <&reg_usdhc1_vmmc>;
-	fsl,sdio-async-interrupt-enabled;
+	vmmc-supply = <&wlan_en>;
 	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
 
-	wifi_wake_host {
-		compatible = "nxp,wifi-wake-host";
+	wlcore: wlcore@2 {
+		compatible = "ti,cc33xx";
+		reg = <2>;
 		interrupt-parent = <&gpio2>;
 		interrupts = <9 IRQ_TYPE_LEVEL_LOW>;
-		interrupt-names = "host-wake";
+	};
+};
+
+&uart1 { 
+	bluetooth {
+		compatible = "ti,cc33xx-bt";
+		cc33xx-supply = <&wlan_en>;
+		max-speed = <115200>;
 	};
 };
diff --git arch/arm64/configs/imx_v8_defconfig arch/arm64/configs/imx_v8_defconfig
index 10979905f06e..d153d9067cc6 100644
--- arch/arm64/configs/imx_v8_defconfig
+++ arch/arm64/configs/imx_v8_defconfig
@@ -367,8 +367,12 @@ CONFIG_USB_NET_MCS7830=m
 CONFIG_USB_NET_CDC_SUBSET=m
 CONFIG_USB_NET_ZAURUS=m
 CONFIG_HOSTAP=y
-CONFIG_WL18XX=m
-CONFIG_WLCORE_SDIO=m
+CONFIG_WL18XX=n
+CONFIG_WLCORE_SDIO=n
+CONFIG_CC33XX=m
+CONFIG_CC33XX_SDIO=m
+CONFIG_BT_TI=m
+CONFIG_BT_TI_UART=m
 CONFIG_XEN_NETDEV_BACKEND=m
 CONFIG_IVSHMEM_NET=y
 CONFIG_INPUT_EVDEV=y
